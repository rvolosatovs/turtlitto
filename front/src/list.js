import React from "react";

/**
 * Show a list of items, but only the items that are enabled.
 * Author: H.E. van der Laan
 *
 * Arguments:
 * - The component to be wrapped
 * - An array of props to be passed to the wrapped components.
 *
 * One WrappedComponent is created for each element of the data array for which the element "enabled" is truthy.
 *
 * Usage warning: Each element of the array data should have an unique element id which will be used as a reconciling hint for React.
 * See https://reactjs.org/warnings/special-props.html and https://reactjs.org/docs/lists-and-keys.html#keys for more details.
 */
export default (WrappedComponent, data) => {
  return (
    <div>
      {data
        .filter(item => {
          return item["enabled"];
        })
        .map(item => {
          return <WrappedComponent key={item.id} {...item} />;
        })}
    </div>
  );
};
