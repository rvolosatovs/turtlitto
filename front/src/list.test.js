import React from "react";
import renderer from "react-test-renderer";
import list from "./list";

const TestComponent = props => {
  return <p>{props.data}</p>;
};

test("Multiple elements are rendered if enabled", () => {
  const component = renderer.create(
    list(TestComponent, [
      { enabled: true, data: 2, id: 2 },
      { enabled: true, data: 1, id: 1 }
    ])
  );
  expect(component.toJSON()).toMatchSnapshot();
});

test("No elements are added when no elements are present", () => {
  const component = renderer.create(list(TestComponent, []));
  expect(component.toJSON()).toMatchSnapshot();
});

test("Elements are only rendered when they are enabled", () => {
  const component = renderer.create(
    list(TestComponent, [
      { enabled: false, data: 2, id: 2 },
      { enabled: true, data: 1, id: 1 }
    ])
  );
  expect(component.toJSON()).toMatchSnapshot();
});
