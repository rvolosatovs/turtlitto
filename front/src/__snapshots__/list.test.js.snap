// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Elements are only rendered when they are enabled 1`] = `
<div>
  <p>
    1
  </p>
</div>
`;

exports[`Multiple elements are rendered if enabled 1`] = `
<div>
  <p>
    2
  </p>
  <p>
    1
  </p>
</div>
`;

exports[`No elements are added when no elements are present 1`] = `<div />`;
